src/app/src/pages/evals/components/EvalsDataGrid.tsx

```diff
@@ -204,6 +229,36 @@ export default function EvalsDataGrid({
 
   const handleCellClick = (params: GridCellParams<Eval>) => onEvalSelected(params.row.evalId);
 
+  /**
+   * Handles the deletion of selected evals.
+   * @returns A promise that resolves when the evals are deleted.
+   */
+  const handleDeleteSelected = async () => {
+    if (rowSelectionModel.length === 0) {
+      return;
+    }
+
+    const confirm = window.confirm(
+      `Are you sure you want to delete ${rowSelectionModel.length} evaluation${
+        rowSelectionModel.length === 1 ? '' : 's'
+      }?`,
+    );
+    if (!confirm) {
+      return;
+    }
+
+    try {
+      await Promise.all(
```

Comment:
Strongly recommend moving the bulk deletion logic into the server instead and doing it sequentially.